webbluetooth.js

```diff
@@ -941,17 +949,40 @@
     errorFn     // Function(String errorMsg)  function called if error occurs
     )
   {
-    init(function() {
-      evothings.ble.stopScan();
-      evothings.ble.startScan(
-          serviceUUIDs,
-        function(deviceInfo) {
-          if (foundFn) { foundFn(createBleatDeviceObject(deviceInfo)); }
-        },
-        function(error) {
-          if (errorFn) { errorFn(error); }
-        });
-      if (completeFn) { completeFn(); }
+    init(function () {
+      let loopIdx = 0;
+
+      const scan = function (idx) {
+        const scanIDs = idx > -1 ? [serviceUUIDs[idx]] : serviceUUIDs;
+        console.log("Scanning for " + scanIDs);
+        evothings.ble.stopScan();
+        evothings.ble.startScan(
+          scanIDs,
+          function (deviceInfo) {
+            adapter.stopScanLoop();
+            if (foundFn) { foundFn(createBleatDeviceObject(deviceInfo)); }
+          },
+          function (error) {
+            adapter.stopScanLoop();
+            if (errorFn) { errorFn(error); }
+          });
+        if (completeFn) { completeFn(); }
+      };
+
+      const scanLoop = function () {
+        scan(loopIdx);
+        adapter.scanLoopTimeout = setTimeout(function () {
+          loopIdx = (loopIdx + 1) % serviceUUIDs.length;
+          scanLoop();
+        }, 2000);
```

Comment:
Isn't 2000 too long if the user has a device way down the list of uuids to search?

Reply:
250ms was too short an interval, and the sensor never showed up. We currently have three different sensors, and two of them share the same ID, so only really 2 sensor types. I will do a couple more tests with intervals and reduce it as far as I can.

Reply:
500ms seemed to work just fine, and should give us plenty of capacity for more sensor types before the delay is noticeable.

Reply:
1000ms was the most reliable interval on the two devices I tested, so will run with this for now.